Extensions/audio_plus.js

```diff
@@ -50,54 +52,162 @@ XKit.extensions.audio_plus = {
 			window.addEventListener("scroll", XKit.extensions.audio_plus.handle_scroll, false);
 			XKit.extensions.audio_plus.create_pop_out_controls();
 		}
+
+		//keep tabs on whether there's a docked video post
+		if (this.can_see_docked_posts) {
+			var targetNode = document.getElementById("right_column");
+			var config = {attributes: true};
+			this.dock_observer.observe(targetNode, config);
+		}
 	},
 
-	create_pop_out_controls: function() {
-		// Create play and pause buttons modeled after the built-in ones
-		var playPause = document.createElement("div");
-		playPause.classList.add("play-pause");
+	dock_observer: new MutationObserver(function(mutations, observer) {
```

Comment:
you can slim down these lines by using arrow function syntax
```js
dock_observer: new MutationObserver((mutations, observer) => {
```
since they're anonymous functions and they don't use their own `this`. furthermore, the `observer` argument is unused, meaning you can remove it and the parenthesis around the now-single argument:
```js
dock_observer: new MutationObserver(mutations => {
```

Reply:
Didn't know I could do that! Looks much nicer, thanks ^__^